client/src/components/Header/HeaderMain/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { SyntheticEvent, useCallback, useState } from 'react';
+import styled from 'styled-components';
+import searchIcon from './searchIcon.svg';
+
+const INPUT_PLACEHOLDER = '검색어를 입력해주세요';
+
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const SearchInput = styled.input`
+  width: 220px;
+  height: 20px;
+  padding: 0 0 5px 3px;
+  background: #fff;
+  border: none;
+  outline: none;
+  border-bottom: 1px solid #333;
+  font-size: 14px;
+
+  :placeholder {
+    color: #999;
+  }
+`;
+
+const SearchButton = styled.button`
+  border: none;
+  background-color: #fff;
+  cursor: pointer;
+
+  img {
+    width: 20px;
+    height: 20px;
+  }
+`;
+
+const SearchBar = (): JSX.Element => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const handleSearchInputChange = useCallback(
+    (e) => setSearchTerm(e.target.value),
+    [setSearchTerm]
+  );
```

Comment:
저는 타입은 항상 명시하는게 좋다고 생각하는데요. 그래서 사실 똑같이 작동하지만 아래와 같이 어떤 이벤트에 어떤 엘리먼트에서 일어나는 이벤트인지 적어주신다면, 좀 더 이해하는데 도움이 될 것 같은데 어떻게 생각하시나요 ?? :)

```suggestion
const handleSearchInputChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => setSearchTerm(e.target.value),
    [setSearchTerm]
  );
```

Reply:
저렇게 타입을 주면 되는지 몰랐네요. 감사합니다. 수정할게요 :)